File: src/lib/scala/Slate.Common/src/main/scala/slate/common/http/common/HttpException.scala
/**
<slate_header>
  author: Kishore Reddy
  url: https://github.com/kishorereddy/scala-slate
  copyright: 2015 Kishore Reddy
  license: https://github.com/kishorereddy/scala-slate/blob/master/LICENSE.md
  desc: a scala micro-framework
  usage: Please refer to license on github for more info.
</slate_header>
  */

package slate.common.http.common

/**
 * Http exception for status
 * @param code   : status code
 * @param msg    : error message
 * @param tag    : optional tag for client reference
 */
case class HttpException(
                          code      : Int,
                          msg       : String,
                          tag       : String = ""
                        )
  extends RuntimeException(code + " Error: " + msg)

File: src/lib/kotlin/slatekit-common/src/main/kotlin/slatekit/common/http/HttpException.kt
/**
 * <slate_header>
 * url: www.slatekit.com
 * git: www.github.com/code-helix/slatekit
 * org: www.codehelix.co
 * author: Kishore Reddy
 * copyright: 2016 CodeHelix Solutions Inc.
 * license: refer to website and/or github
 * about: A tool-kit, utility library and server-backend
 * mantra: Simplicity above all else
 * </slate_header>
 */

package slatekit.common.http

/**
 * Http exception for status
 * @param code   : status code
 * @param msg    : error message
 * @param tag    : optional tag for client reference
 */
data class HttpException(
        val code: Int,
        val msg: String,
        val tag: String = ""
)
    : RuntimeException(code.toString() + " Error: " + msg)
